packages/clerk-js/src/ui/utils/colors/modern.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Modern CSS-based color manipulation utilities
+ * Uses color-mix() and relative color syntax when supported
+ */
+
+import { cssSupports } from '../cssSupports';
+import { hasModernColorSupport as hasModernColorSupportCached } from './cache';
+
+/**
+ * Modern CSS-based color manipulation utilities
+ * Uses color-mix() and relative color syntax when supported
+ */
+export const modernColors = {
+  /**
+   * Lightens a color by a percentage using modern CSS
+   */
+  lighten: (color: string | undefined, percentage = 0): string | undefined => {
+    if (!color) return undefined;
+
+    if (cssSupports.relativeColorSyntax()) {
+      // Use relative color syntax for precise lightness control
+      const lightnessIncrease = percentage * 100; // Convert to percentage
+      return `hsl(from ${color} h s calc(l + ${lightnessIncrease}%))`;
+    }
+
+    if (cssSupports.colorMix()) {
+      // Use color-mix as fallback
+      const mixPercentage = Math.min(percentage * 100, 95); // Cap at 95%
+      return `color-mix(in srgb, ${color}, white ${mixPercentage}%)`;
+    }
+
+    return color; // Return original if no modern CSS support
+  },
+
+  /**
+   * Makes a color transparent by a percentage using modern CSS
+   */
+  makeTransparent: (color: string | undefined, percentage = 0): string | undefined => {
+    if (!color || color.toString() === '') return undefined;
+
+    if (cssSupports.colorMix()) {
+      // Use color-mix with transparent
+      const alphaPercentage = Math.max((1 - percentage) * 100, 5); // Minimum 5% opacity
```

Comment:
i think we can do this in css yeah? probably doesn't matter, but wanted to point it out